share/completions/rustc.fish

```diff
@@ -49,13 +49,13 @@ for line in $rust_docs
 end
 
 set -l rust_docs (rustc -W help  \
-    | egrep \
-        '(\s+)(.+)(\s+)(allow|warn|deny|forbid)(\s+){2}([^\n]+)' \
+    | string match -r \
+        '(?:\s+)(?:.+)(?:\s+)(?:allow|warn|deny|forbid)(?:\s+){2}(?:[^\n]+)' \
     | string replace -r -i \
```

Comment:
I feel like there should be some means with just `string replace` and the pcre2 syntax if I better understood regular expressions I could do lines 52-54 this with one string replace... I can't nail it.